L1Trigger/L1THGCal/python/customClustering.py

```diff
@@ -76,58 +76,101 @@ def custom_3dclustering_dbscan(process,
 
 
 def custom_3dclustering_histoMax(process,
-        distance = 0.01,
+        distance = 0.03,
         nBins_R = 36,
         nBins_Phi = 216,
         binSumsHisto = binSums,                        
+        seed_threshold = 10,
         ):
     parameters_c3d = process.hgcalBackEndLayer2Producer.ProcessorParameters.C3d_parameters
     parameters_c3d.dR_multicluster = cms.double(distance)
     parameters_c3d.nBins_R_histo_multicluster = cms.uint32(nBins_R)
     parameters_c3d.nBins_Phi_histo_multicluster = cms.uint32(nBins_Phi)
     parameters_c3d.binSumsHisto = binSumsHisto
+    parameters_c3d.threshold_histo_multicluster = seed_threshold
     parameters_c3d.type_multicluster = cms.string('HistoMaxC3d')
     return process
 
+def custom_3dclustering_histoSecondaryMax(process,
+        distance = 0.03, 
+        threshold = 5.,
+        nBins_R = 36,
+        nBins_Phi = 216,
+        binSumsHisto = binSums,
+        ):
+    parameters_c3d = process.hgcalBackEndLayer2Producer.ProcessorParameters.C3d_parameters
+    parameters_c3d.dR_multicluster = cms.double(distance)
+    parameters_c3d.nBins_R_histo_multicluster = cms.uint32(nBins_R)
+    parameters_c3d.nBins_Phi_histo_multicluster = cms.uint32(nBins_Phi)
+    parameters_c3d.binSumsHisto = binSumsHisto
+    parameters_c3d.threshold_histo_multicluster = cms.double(threshold)
+    parameters_c3d.type_multicluster = cms.string('HistoSecondaryMaxC3d')
+    return process
+
+dr_layerbylayer = ([0] + # no layer 0
+        [0.015]*7 + [0.020]*7 + [0.030]*7 + [0.040]*7 + # EM
```

Comment:
repeated magic numbers 7, 6, 12 should be named constants
(easier to change them consistently in case the geometry changes)

Reply:
Is there some particular rule in CMSSW for constant variables except the standard capitalization.

Reply:
I don't think there is any special naming rule for constants. We just prefer to avoid (unlabeled) magic numbers, because they can lead to inconsistencies.